lib/Service/OpmlService.php

```diff
@@ -57,13 +60,14 @@ public function import(string $userId, string $data): bool
     {
         list($folders, $feeds) = $this->importer->import($userId, $data);
 
+        $error = false;
         $folderEntities  = [];
         $dbFolders = $this->folderService->findAllForUser($userId);
         foreach ($folders as $folder) {
             $existing = array_filter($dbFolders, fn(Folder $dbFolder) => $dbFolder->getName() === $folder['name']);
 
             if (count($existing) > 0) {
-                $folderEntities[$folder['name']] = $existing[0];
+                $folderEntities[$folder['name']] = reset($existing);
```

Comment:
Why reset?

Reply:
array_filter keeps the array index of the found object, there for you need to reset it.

Reply:
`array_pop` just gives you the next item. That might be better.